lib/NamedModulesPlugin.js

```diff
@@ -12,13 +12,31 @@ class NamedModulesPlugin {
 	apply(compiler) {
 		compiler.hooks.compilation.tap("NamedModulesPlugin", compilation => {
 			compilation.hooks.beforeModuleIds.tap("NamedModulesPlugin", modules => {
+				let namedModules = {};
+
 				for (const module of modules) {
 					if (module.id === null && module.libIdent) {
 						module.id = module.libIdent({
 							context: this.options.context || compiler.options.context
 						});
 					}
+
+					if (module.id) {
+						(namedModules[module.id] && namedModules[module.id].push(module)) ||
+							(namedModules[module.id] = Array.of(module));
+					}
 				}
+
+				Object.keys(namedModules).forEach(key => {
+					const namedModule = namedModules[key];
+					if (namedModule.length > 1) {
+						namedModule.forEach(module => {
+							if (module.issuer && module.issuer.id) {
```

Comment:
Instead resolve the conflict by appending a hash digest of `module.identifier()` as query argument:

i. e. `./lib/module.js` -> `./lib/module.js?4db2`

Please also shorten the `identifier()` with a `requestShortener` to be context-independent.

You can steal this from here:

https://github.com/webpack/webpack/blob/master/lib/ModuleFilenameHelpers.js#L44-L48
https://github.com/webpack/webpack/blob/master/lib/ModuleFilenameHelpers.js#L83
https://github.com/webpack/webpack/blob/master/lib/ModuleFilenameHelpers.js#L89